maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/SurefireHelper.java

```diff
@@ -142,7 +142,10 @@ public static void reportExecution( SurefireReportParameters reportParameters, R
                                         PluginConsoleLogger log, Exception firstForkException )
         throws MojoFailureException, MojoExecutionException
     {
-        if ( firstForkException == null && !result.isTimeout() && result.isErrorFree() )
+        boolean isError = firstForkException != null || result.isTimeout() || !result.isErrorFree();
+        boolean isFlaky = reportParameters.getFailOnFlakeCount() > 0
```

Comment:
this stores more than whether we're flaky, rather something like: `isFlakyError`, right?

Reply:
minor i would say